docs/contracts/inheritance.rst

```diff
@@ -108,93 +115,170 @@ Details are given in the following example.
     // If a constructor takes an argument, it needs to be
     // provided in the header or modifier-invocation-style at
     // the constructor of the derived contract (see below).
-    contract PriceFeed is Owned, Destructible, Named("GoldFeed") {
+    contract PriceFeed is Owned, Pausable, Named("GoldFeed") {
+        uint info;
+
         function updateInfo(uint newInfo) public {
             if (msg.sender == owner) info = newInfo;
         }
 
         // Here, we only specify `override` and not `virtual`.
         // This means that contracts deriving from `PriceFeed`
-        // cannot change the behavior of `destroy` anymore.
-        function destroy() public override(Destructible, Named) { Named.destroy(); }
+        // cannot change the behavior of `pause` anymore.
+        function pause() public override(Pausable, Named) { Named.pause(); }
         function get() public view returns(uint r) { return info; }
-
-        uint info;
     }
 
-Note that above, we call ``Destructible.destroy()`` to "forward" the
-destruction request. The way this is done is problematic, as
+Note that above, we call ``Pausable.pause()`` to "forward" the
+pause request. The way this is done is problematic, as
 seen in the following example:
 
 .. code-block:: solidity
 
     // SPDX-License-Identifier: GPL-3.0
     pragma solidity >=0.7.0 <0.9.0;
-    // This will report a warning due to deprecated selfdestruct
 
-    contract owned {
-        constructor() { owner = payable(msg.sender); }
+    contract Owned {
         address payable owner;
+        constructor() { owner = payable(msg.sender); }
     }
 
-    contract Destructible is owned {
-        function destroy() public virtual {
-            if (msg.sender == owner) selfdestruct(owner);
+    contract Pausable is Owned {
+        bool private _paused;
+
+        event Paused();
+        event Unpaused();
+
+        constructor() { _paused = false; }
+
+        function paused() public view virtual returns (bool) {
+            return _paused;
+        }
+
+        function pause() virtual public {
+            if (msg.sender == owner && !_paused)
+            {
+                emit Paused();
+                _paused = true;
+            }
+        }
+
+        function unpause() virtual public {
+            if (msg.sender == owner && _paused)
+            {
+                emit Unpaused();
+                _paused = false;
+            }
         }
     }
```

Comment:
This was meant to illustrate the inheritance and I think that the new example being this much larger makes this section lose focus a bit. IMO it would be better to still keep it concise. How about we drop `unpause()`, `paused()` and `_paused` and just have a single function that emits the `Paused` event, to replace `destroy()`?

Same for a similar contract below.

Reply:
But I think we should at least keep the `_paused` variable so that the contract can be "paused," for the sake of completeness in the example

Reply:
Actually, rethinking about it, you are completely right. I changed the example ;)

Reply:
I mean, I'm talking only about these short contracts that illustrate inheritance. Those were meant to be short because they illustrate a specific point. There's a bigger pausable example above them and there having all the elements is perfectly fine and even recommended.